CMakeLists.txt

```diff
@@ -48,16 +47,27 @@ endif()
 # Add the FINUFFT project using the vendored version
 add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/vendor/finufft")
 
-# Find pybind11
-set(PYBIND11_NEWPYTHON ON)
-find_package(pybind11 CONFIG REQUIRED)
+# Find Python
+find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
+
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# Find nanobind
+execute_process(
+    COMMAND "${Python_EXECUTABLE}" -m nanobind --cmake_dir
+    OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE NB_DIR)
+list(APPEND CMAKE_PREFIX_PATH "${NB_DIR}")
```

Comment:
I've never actually had to do this part with nanobind, despite what the documentation says. I think it's fine to leave it since the documentation encourages it, but the scikit-build-core documentation says "Nanobind places its config file such that CMake can find it from site-packages." (https://scikit-build-core.readthedocs.io/en/latest/getting_started.html#cmake-file). Likewise, the "official" example from the nanobind folks doesn't do this discovery manually: https://github.com/wjakob/nanobind_example/blob/master/CMakeLists.txt

Not suggesting we need to change this, just noting it as a curiosity!